frontend/frontend.go

```diff
@@ -142,11 +154,18 @@ func (f *Frontend) NotFound(writer http.ResponseWriter, request *http.Request) {
 }
 
 func (f *Frontend) HealthzReady(writer http.ResponseWriter, request *http.Request) {
+	var healthStatus float64
 	if f.CheckReady() {
 		writer.WriteHeader(http.StatusOK)
+		healthStatus = 1.0
 	} else {
 		writer.WriteHeader(http.StatusInternalServerError)
+		healthStatus = 0.0
 	}
+
+	(*PrometheusEmitter).EmitGauge(f.metrics, "frontend.health", healthStatus, map[string]string{
```

Comment:
I am not understanding something, probably pure ignorance on my side 🙂 .

Why are we passing as the first parameter of _EmitGauge_ the field _f.metrics_ (something of type _*PrometheusEmitter_) when it is expecting a parameter of type string?

Also, what is the expression 
_(*PrometheusEmitter).EmitGauge(...)_
doing here? Are we calling the method _EmitGauge_ in a just in time PrometheusEmitter instantiated ? I think this line could be something like:
```go
name := "a good name for this"
f.emitter.EmitGauge(name, "frontend.health", healthStatus, map[string]string{
		"endpoint": "/healthz/ready",
})
```
where f.emitter is the field I discussed in my previous comments (of type metrics.Emitter)